Release 3.2.0-BETA.2 (#1818)
microsoft/ApplicationInsights-Java: CHANGELOG.md

# CHANGELOG

> # Version 3.2.0-BETA.3

Reviewer: this is BETA.2
```suggestion
# Version 3.2.0-BETA.2
```

Reviewer 2: yeah, we didn't release beta.2.  @kryalama  please reuse this draft: https://github.com/microsoft/ApplicationInsights-Java/releases/tag/untagged-969a656e91fa14295250

> * Add Spring integration (preview).

Reviewer: ```suggestion
* Add Spring Integration instrumentation (preview).
```

> * Add thread name to log capture. Resolves [#1699](https://github.com/microsoft/ApplicationInsights-Java/issues/1699)

Reviewer 2: can we remove Resolves everywhere in this pr.  put link before the period.

> * Use azure monitor exporter instead of Open Telemetry exporter.
> * Add data truncation and sanitization to telemetry data.

Reviewer: I don't think there's any end-user facing changes here, in which case no need to mention
```suggestion
```

* Support W3C baggage propagation.

> * Update App Services internal diagnostics comments.

Reviewer: not end-user facing
```suggestion
```

* Disable legacy Request-Id propagation by default.
* Handle partial content success due to Stamp specific redirects. Resolves [#1674](https://github.com/microsoft/ApplicationInsights-Java/issues/1674)
* Log controlled warning when telemetry truncated. Resolves [#1021](https://github.com/microsoft/ApplicationInsights-Java/issues/1021)

> * Support system properties for the main "coordinates". Resolves [#1684](https://github.com/microsoft/ApplicationInsights-Java/issues/1684)

Reviewer: ```suggestion
* Support system properties for the connection string, role name and role instance. Resolves [#1684](https://github.com/microsoft/ApplicationInsights-Java/issues/1684)
```

> * Support system properties for the main "coordinates". Resolves [#1684](https://github.com/microsoft/ApplicationInsights-Java/issues/1684)

Reviewer 2: I will be more specific about this 'coordinates', list them please.

> * Small fixes to 2.x interop and exporter code cleanup.

Reviewer: ```suggestion
* Small fixes to 2.x interop.
```

> * Persist regular telemetries on disk inorder to retry when failed to send telemetry.

Reviewer: this is not a change (well, I guess it's a change from BETA.1, but just back to normal)
```suggestion
```

> * Persist regular telemetries on disk inorder to retry when failed to send telemetry.

Reviewer 2: Add disk persistence?
